### PR TITLE
Header & Footer Plaintext Support 

### PR DESCRIPTION
**Pull request by davidcui-amzn**
*Issue #, if available:*
N/A
*Description of changes:*
Header and footer have plain-text support and will display in the report header and footer as plaintext.

`TODO`: Add formatting to the report to clean up the display of the header and footer with the report. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by joshuali925 — APPROVED** [review]

File: kibana-reports/public/components/report_definitions/create/create_report_definition.tsx [hunk 1]
```diff
@@ -35,6 +35,8 @@ interface reportParamsType {
   report_name: string;
   report_source: string;
   description: string;
+  header: string;
+  footer: string;
```
Comment (zhongnansu) [inline kibana-reports/public/components/report_definitions/create/create_report_definition.tsx:39]: This is only for create report definition, also need to add support for edit report definition

File: kibana-reports/server/model/index.ts [hunk 2]
```diff
@@ -113,7 +113,8 @@ export const reportDefinitionSchema = schema.object({
       schema.literal(REPORT_TYPE.savedSearch),
     ]),
     description: schema.string(),
-
+    header: schema.string(),
+    footer: schema.string(),
```
Comment (zhongnansu) [inline kibana-reports/server/model/index.ts:117]: If you add the fields here, which means header and footer are for all types of report. I don't think we need header/footer for csv report, can you confirm? If so, add header/footer fields inside the  `visualReportSchema`. If these fields are optional, use `header: schema.maybe(schema.string())`

**Review by zhongnansu — COMMENTED** [review]
Please address the comments

File: kibana-reports/public/components/report_definitions/report_settings/report_settings.tsx [hunk 3]
```diff
@@ -207,6 +207,16 @@ export function ReportSettings(props: ReportSettingProps) {
       'write' | 'preview'
     >('write');
 
+    const handleHeader = (e) => {
+      setHeader(e);
+      reportDefinitionRequest.report_params.header = e;
+    };
```
Comment (zhongnansu) [inline kibana-reports/public/components/report_definitions/report_settings/report_settings.tsx:213]: Any input validation for header and footer? Can user choose not to add any header and footer at all?

**Comment by zhongnansu** [comment]
I didn't see any code change in `report detail` or `report definition detail`. It may not be displayed on those two pages.

File: kibana-reports/public/components/report_definitions/create/create_report_definition.tsx (hunk 1 above)
Comment (davidcui-amzn) [reply kibana-reports/public/components/report_definitions/create/create_report_definition.tsx:39]: Good catch, will add

File: kibana-reports/public/components/report_definitions/report_settings/report_settings.tsx (hunk 3 above)
Comment (davidcui-amzn) [reply kibana-reports/public/components/report_definitions/report_settings/report_settings.tsx:213]: Yes, header and footer can be blank- not sure what input validation to include, hasn't been discussed before.

**Review by zhongnansu — APPROVED** [review]

**Pull request merged**